index.js

```diff
@@ -49,6 +49,22 @@ module.exports = ({ types: t }) => {
     }
   };
 
+  const isModuleOrExportsInjected = (dependencyList, factoryArity) => {
```

Comment:
I would break this into two sub functions so that you can give a name to each concern. It took me a while to figure out which half was about which concern. You could still keep this function as a wrapper, but it cold then just be a ternary that calls out to the other two.